File: back/game.cpp
#include <iostream>
#include <vector>
#include <ctime>   // For time()
#include <cstdlib> // For rand() and srand()

using namespace std;

char board[9] = {'0', '1', '2', '3', '4', '5', '6', '7', '8'};

void drawBoard()
{
    cout << "\n";
    for (int i = 0; i < 9; i += 3)
    {
        cout << " | " << board[i] << " | " << board[i + 1] << " | " << board[i + 2] << " | \n";
    }
}

bool checkWin()
{
    int wins[8][3] = {{0, 1, 2}, {3, 4, 5}, {6, 7, 8}, {0, 3, 6}, {1, 4, 7}, {2, 5, 8}, {0, 4, 8}, {2, 4, 6}};
    for (int i = 0; i < 8; i++)
    {
        if (board[wins[i][0]] == board[wins[i][1]] && board[wins[i][1]] == board[wins[i][2]])
            return true;
    }
    return false;
}

int main()
{
    int mode, choice;
    char player = 'X';

    cout << "--- C++ Tic Tac Toe ---\n";
    cout << "Select Mode: (1) PvP | (2) vs Computer: ";
    cin >> mode;

    for (int i = 0; i < 9; i++)
    {
        drawBoard();

        if (mode == 2 && player == 'O')
        {
            cout << "Computer is thinking...\n";
            srand(time(0)); // Seed the random number generator
            do
            {
                choice = rand() % 9; // Generate random number 0-8
            } while (board[choice] == 'X' || board[choice] == 'O');
        }
        else
        {
            cout << "Player " << player << ", enter cell: ";
            cin >> choice;
            if (board[choice] == 'X' || board[choice] == 'O')
            {
                cout << "Spot taken! Try again.\n";
                i--;
                continue;
            }
        }

        board[choice] = player;
        if (checkWin())
        {
            drawBoard();
            cout << "🎉 Player " << player << " wins!\n";
            return 0;
        }
        player = (player == 'X') ? 'O' : 'X';
    }

    drawBoard();
    cout << "🤝 It's a draw!\n";
    return 0;
}

File: game.py
import random

def print_board(board):
    for i in range(0, 9, 3):
        print(f"| {board[i]} | {board[i+1]} | {board[i+2]} |")

def check_winner(board):
    wins = [(0,1,2), (3,4,5), (6,7,8), (0,3,6), (1,4,7), (2,5,8), (0,4,8), (2,4,6)]
    for w in wins:
        # Check if the three spots are the same and NOT empty
        if board[w[0]] == board[w[1]] == board[w[2]] != " ":
            return True
    return False

# --- GAME SETUP ---
board = [" " for _ in range(9)]
print("--- Welcome to Tic Tac Toe ---")
mode = input("Select Mode: (1) PvP | (2) vs Computer: ")
turn = "X"

for i in range(9):
    print_board(board)
    
    # If it's Computer's turn in mode 2
    if mode == "2" and turn == "O":
        print("Computer is thinking...")
        move = random.choice([idx for idx, val in enumerate(board) if val == " "])
    else:
        try:
            move = int(input(f"Player {turn}, enter position (0-8): "))
            if board[move] != " ":
                print("Spot taken! Try again.")
                continue
        except (ValueError, IndexError):
            print("Invalid input! Use numbers 0-8.")
            continue

    board[move] = turn
    
    if check_winner(board):
        print_board(board)
        print(f"🎉 Player {turn} wins!")
        break
    
    turn = "O" if turn == "X" else "X"
else:
    print_board(board)
    print("🤝 It's a draw!")